app/gui/qt/mainwindow.cpp

```diff
@@ -1262,6 +1262,13 @@ void MainWindow::load()
     delete load_from_dialog;
 }
 
+
+void MainWindow::launch_share(const QString share_filename)
+{
+    // Loads "share_filename" directly into the workspace
+    getCurrentWorkspace()->setText(share_filename);
```

Comment:
I'm a bit confused - surely this will just fill the workspace with something like

```
my_share.spi
```

rather than reading the file and populating?